### PR TITLE
Bump istio to 1.18.1

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Bump istio to 1.18.1

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,9 +19,9 @@ require (
 	gitlab.com/rodrigoodhin/gocure v0.0.0-20220718065339-f14dfe79276a
 	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1
 	golang.org/x/time v0.3.0
-	google.golang.org/protobuf v1.31.0
-	istio.io/api v0.0.0-20230621030812-36698ea550c8
-	istio.io/client-go v1.18.1-0.20230621031510-0789e128a186
+	google.golang.org/protobuf v1.30.0
+	istio.io/api v0.0.0-20230713061407-06047cce866f
+	istio.io/client-go v1.18.1
 	istio.io/istio v0.0.0-20230621044911-6f0d7ae668c8
```
Comment (triffer) [inline go.mod:25]: IS there a reason why we didn't update `istio.io/istio`?

File: go.mod [hunk 2]
```diff
@@ -19,9 +19,9 @@ require (
 	gitlab.com/rodrigoodhin/gocure v0.0.0-20220718065339-f14dfe79276a
 	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1
 	golang.org/x/time v0.3.0
-	google.golang.org/protobuf v1.31.0
```
Comment (triffer) [inline go.mod:22]: What is the reason for the downgrade?

File: docs/release-notes/0.2.1.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+## New features
+Updated Istio version to 1.18.1
```
Comment (nataliasitko) [inline docs/release-notes/0.2.1.md:2]:
```suggestion
- Update the Istio version to 1.18.1
```

File: go.mod (hunk 1 above)
Comment (triffer) [reply go.mod]: 😄 I remembered that we wanted to write a doc for updating Istio library when we do it the next time. 😏

**Review by triffer — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**